src/main/api/2/studio-api-2.yaml

```diff
@@ -1456,6 +1460,235 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /repository/add_remote:
+    post:
+      tags:
+        - repository
+      summary: Add remote repository to the site content repository
+      operationId: addRemoteRepository
+      requestBody:
+        description: Remote repository entity
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RemoteRepository'
+      responses:
+        '201':
+          description: Added
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  response:
+                    $ref: '#/components/schemas/ApiResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '409':
+          $ref: '#/components/responses/Conflict'
+        '500':
+          $ref: '#/components/responses/InternalServerError'          
+
+  /repository/pull_from_remote:
+    post:
+      tags:
+        - repository
+      summary: Pull content from remote repository to site content repository
+      operationId: pullFromRemoteRepository
+      requestBody:
+        description: pull from remote repository request body
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                siteId:
+                  type: string
+                  description: Site ID
+                remoteName:
+                  type: string
+                  description: Remote repository name to pull from
+                remoteBranch:
+                  type: string
+                  description: Remote repository branch to pull from
```

Comment:
Add ability to specify merge strategy: https://www.git-scm.com/docs/git-pull#Documentation/git-pull.txt--Xltoptiongt